### PR TITLE
Various support changes and cleanups for the jiterpreter PR

### PR DESCRIPTION
**Pull request by kg**
* Introduce mono_interp_is_method_multicastdelegate_invoke
* Add interpreter opcodes for Math.Min, Math.Max, and Math.Abs
* Make it easy to configure AOT+trimming for browser-bench to do comparison tests between AOT and interp
* Add missing `td->cbb` update in interpreter transform
* Add `mono_wasm_array_length_ref` missing safe wasm API
* Optimize out linker placeholder wrappers that were slowing down most C->TS calls
* Generate warning messages for missing cwraps imports from C
* Simplify/optimize _zero_region
Acts as support for #76477

### CODE REVIEW

File: src/mono/mono/mini/interp/interp.c [hunk 1]
```diff
@@ -2145,6 +2145,12 @@ typedef struct {
 	gpointer *many_args;
 } InterpEntryData;
 
+static gboolean
+mono_interp_is_method_multicastdelegate_invoke (MonoMethod *method)
```
Comment (BrzVlad) [inline src/mono/mono/mini/interp/interp.c:2149]: I would drop the `mono` prefix for static methods. Maybe even the `interp` prefix for such small methods that are not really interp related.

File: src/mono/mono/mini/interp/interp.c [hunk 2]
```diff
@@ -3530,6 +3536,56 @@ static long total_executed_opcodes;
 
 #define LOCAL_VAR(offset,type) (*(type*)(locals + (offset)))
 
+/*
+ * Custom C implementations of the min/max operations for float and double.
+ * We cannot directly use the C stdlib functions because their semantics do not match
+ *  the C# methods in System.Math, but having interpreter opcodes for these operations
+ *  improves performance for FP math a lot in some cases.
+ */
+static float min_f (float lhs, float rhs) {
```
Comment (BrzVlad) [inline src/mono/mono/mini/interp/interp.c:3545]: formatting:
```
static float
min_f (float lhs, float rhs)
{
...
```

File: src/mono/mono/mini/interp/interp.c (hunk 1 above)
Comment (kg) [reply src/mono/mono/mini/interp/interp.c:2149]: it becomes non-static for the jiterpreter once it starts being used by it

File: src/mono/mono/mini/interp/interp.c (hunk 2 above)
Comment (kg) [reply src/mono/mono/mini/interp/interp.c:3545]: thanks, will fix

File: src/mono/mono/mini/interp/interp.c [hunk 3]
```diff
@@ -3530,6 +3536,56 @@ static long total_executed_opcodes;
...
+	if (isnan(lhs))
```
Comment (BrzVlad) [inline src/mono/mono/mini/interp/interp.c:3546]: it would probably be cleaner to use the mono_ versions of these math methods

Comment (kg) [reply src/mono/mono/mini/interp/interp.c:3546]: if they match the .NET System.Math semantics, I'll use them. Will look into it

Comment (kg) [reply src/mono/mono/mini/interp/interp.c]: are there mono_ versions of min and max? i didn't find any.

Comment (BrzVlad) [reply src/mono/mono/mini/interp/interp.c]: I meant for `isnan` etc

File: src/mono/sample/wasm/browser-bench/Wasm.Browser.Bench.Sample.csproj [hunk 4]
```diff
@@ -6,6 +6,14 @@
     <WasmMainJSPath>main.js</WasmMainJSPath>
     <SuppressTrimAnalysisWarnings>true</SuppressTrimAnalysisWarnings>
     <EmitCompilerGeneratedFiles>true</EmitCompilerGeneratedFiles>
+    <EnableAOTAndTrimming>true</EnableAOTAndTrimming>
```
Comment (pavelsavara) [inline src/mono/sample/wasm/browser-bench/Wasm.Browser.Bench.Sample.csproj:9]: @radekdoulik would make your perf testing all AOT ?

File: src/mono/wasm/runtime/dotnet.d.ts [hunk 5]
```diff
@@ -3,7 +3,7 @@
 //!
 //! This is generated file, see src/mono/wasm/runtime/rollup.config.js
 
-//! This is not considered public API with backward compatibility guarantees. 
+//! This is not considered public API with backward compatibility guarantees.
```
Comment (pavelsavara) [inline src/mono/wasm/runtime/dotnet.d.ts:6]: please revert or fix the way how it is generated

**Review by pavelsavara — APPROVED** [review]
Otherwise JS side LGTM

File: src/mono/sample/wasm/browser-bench/Wasm.Browser.Bench.Sample.csproj (hunk 4 above)
Comment (kg) [reply src/mono/sample/wasm/browser-bench/Wasm.Browser.Bench.Sample.csproj:9]: ah, good point, the other projects I did this to had AOT on originally but this didn't. I'll change it back.

File: src/mono/mono/mini/interp/interp.c [hunk 6]
```diff
@@ -3530,6 +3536,60 @@ static long total_executed_opcodes;
...
 #define LOCAL_VAR(offset,type) (*(type*)(locals + (offset)))
 
+/*
+ * Custom C implementations of the min/max operations for float and double.
+ * We cannot directly use the C stdlib functions because their semantics do not match
+ *  the C# methods in System.Math, but having interpreter opcodes for these operations
+ *  improves performance for FP math a lot in some cases.
+ */
+static float
+min_f (float lhs, float rhs) {
```
Comment (BrzVlad) [inline src/mono/mono/mini/interp/interp.c:3546]: parenthesis on new line

**Pull request merged**